server/src/main/java/org/apache/druid/server/coordinator/CuratorLoadQueuePeon.java

```diff
@@ -431,34 +392,28 @@ public int getType()
       return type;
     }
 
-    public SegmentId getSegmentId()
+    public String getSegmentIdentifier()
```

Comment:
I don't see a point of `String`-ifying API. Return a `SegmentId` object and let the users chain-call `toString()` if they need.

Reply:
SegmentHolder is an inner class and almost all the callers who need segmentIdentifier need the string version of it. So it made more sense to me to just return the String instead of having callers call toString() on segmentId.